src/main/java/com/artipie/vertx/VertxSliceServer.java

```diff
@@ -127,18 +128,24 @@ public void close() {
                 req.headers(),
                 req.toFlowable().map(buffer -> ByteBuffer.wrap(buffer.getBytes()))
             ).send(
-                (code, headers, body) -> {
+                (status, headers, body) -> {
+                    final int code = Integer.parseInt(status.code());
                     final HttpServerResponse response = req.response().setStatusCode(code);
                     for (final Map.Entry<String, String> header : headers) {
                         response.putHeader(header.getKey(), header.getValue());
                     }
                     response.setChunked(true);
+                    final CompletableFuture<Void> promise = new CompletableFuture<>();
                     Flowable.fromPublisher(body).map(
                         buf -> {
                             final byte[] bytes = new byte[buf.remaining()];
                             buf.get(bytes);
                             return Buffer.buffer(bytes);
-                        }).subscribe(response.toSubscriber());
+                        })
+                        .doOnComplete(() -> promise.complete(null))
```

Comment:
@olegmoz I would suggest to hold some data in the promise, maybe the body content, or the HttpServerResponse altogether. That will give some information to the promise handler, and will avoid for us to have `null` values.

Reply:
@HDouss got rid of null